tests/io/array_io_tests.f90

```diff
@@ -1,27 +1,31 @@
-#include "ekat_config.f"
-
 module unit_test_mod
   implicit none
 contains
 
   function test_array_io() result(nerr) bind(c)
-    use array_io_mod
+    ! Precision independent use clauses
     use iso_c_binding
+    use array_io_mod, only: array_io_file_exists
 
 #ifdef EKAT_DOUBLE_PRECISION
-    integer, parameter :: c_real = c_double
-#else
-    integer, parameter :: c_real = c_float
+    use iso_c_binding, only: c_real=>c_double
+    use array_io_mod, only: array_io_write=>array_io_write_double, array_io_read=>array_io_read_double
+#elif defined(EKAT_SINGLE_PRECISION)
+    use iso_c_binding, only: c_real=>c_float
+    use array_io_mod, only: array_io_write=>array_io_write_float, array_io_read=>array_io_read_float
 #endif
 
-
     integer(kind=c_int) :: nerr
     integer ::  i, j
     real(kind=c_real), target :: a(10,3), b(10,3)
     logical :: ok
 
     character(kind=c_char, len=128), parameter :: &
-         filename = c_char_"unit_test_f90_array_io.dat"//C_NULL_CHAR
+#ifdef EKAT_DOUBLE_PRECISION
+         filename = c_char_"unit_test_f90_array_io_dp.dat"//C_NULL_CHAR
```

Comment:
Do you intend this as a C string in Fortran, or a Fortran string using C-compatible characters? My understanding is that it's the latter. Last I checked, C strings map to Fortran arrays of character(kind=c_char,len=1). But if you've had success using this, that would be interesting.

Reply:
It's the latter. A Fortran array of T is interpreted as a ptr to T in C, so this works. However, notice that in the interoperable subroutines that use such strings, you need to declare them with an unspecified length (see ekat_array_io_mod.f90).

Reply:
Argh, that file is still called `array_io_mod.f90`, while I thought I prepended `ekat_` to all files...